src/main/java/karen/Parser.java

```diff
@@ -0,0 +1,223 @@
+            break;
+        case "unmark":
+            cmd = new ModifyCommand(Integer.valueOf(matcher.group("index")) - 1, ModifyType.UNMARK);
+            break;
+        default:
+            return new InvalidCommand("How did this even get here.");
+        }
+        return cmd;
+    }
+
+    /**
+     * Creates DeleteCommand object by parsing parameters to get (1-based index) of Tasks
+     * to delete.
+     * @param keyWord first word of input
+     * @param fullInput full input from user
+     * @return DeleteCommand object
+     */
+    private Command prepareDelete(String keyWord, String fullInput) {
+        final Matcher matcher = INDEX_FORMAT.matcher(fullInput);
+        if (!matcher.matches()) {
+            return new InvalidCommand("Incorrect arguments passed into delete command");
+        }
+
+        try {
+            return new DeleteCommand(Integer.valueOf(matcher.group("index")) - 1);
+        } catch (IllegalStateException err) {
+            return new InvalidCommand("Missing arguments for delete command");
+        }
+    }
+
+    /**
+     * Creates FindCommand object by parsing parameters to get keyTerm to search with
+     * @param keyWord first word of input
+     * @param fullInput full input from user
+     * @return FindCommand object
+     */
+    private Command prepareFind(String keyWord, String fullInput) {
+        final Matcher matcher = FIND_FORMAT.matcher(fullInput.trim());
+        if (!matcher.matches()) {
+            return new InvalidCommand("'find' command needs a term to even search with");
+        }
+
+        return new FindCommand(matcher.group("keyTerm"));
+    }
+
+    /**
+     * Creates Command object from the full user input
+     * @param fullInput full input from user
+     * @return Command object
+     */
+    private Command createCommand(String fullInput) {
+        // extract first word
+        final Matcher matcher = KEYWORD_FORMAT.matcher(fullInput);
+        if (!matcher.matches()) {
+            return new InvalidCommand(NA_MESSAGE);
+        }
+
+        String keyWord = matcher.group("keyWord");
+        Command getCommand;
+
+        switch (keyWord) {
+        case "list":
+            getCommand = new ListCommand();
+            break;
+        case "bye":
+            getCommand = new ByeCommand();
+            break;
+        case "todo":
+        case "deadline":
+        case "event":
+            getCommand = prepareAdd(keyWord, fullInput);
+            break;
+        case "mark":
+        case "unmark":
+            getCommand = prepareModify(keyWord, fullInput);
+            break;
+        case "delete":
+            getCommand = prepareDelete(keyWord, fullInput);
+            break;
+        case "find":
+            getCommand = prepareFind(keyWord, fullInput);
+            break;
+        default:
+            getCommand = new InvalidCommand(NA_MESSAGE);
+            break;
+        };
+        return getCommand;
+    }
```

Comment:
Great use of switch-case statement to account for the keywords Karen recognizes 👍 
Perhaps Karen could be more helpful to her users when they enter an InvalidCommand by listing the Commands that she understands? (list, bye, todo etc.) 😅

Reply:
Thanks, that's a great point I overlooked